app/ui/browser-modern/views/browser/tabbar/tab-contents.jsx

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import React, { Component, PropTypes } from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import { connect } from 'react-redux';
+
+import Style from '../../../../shared/style';
+import FittedImage from '../../../../shared/widgets/fitted-image';
+
+import PageState from '../../../model/page-state';
+import * as PagesSelectors from '../../../selectors/pages';
+
+const TAB_CONTENTS_STYLE = Style.registerStyle({
+  flex: 1,
+  alignItems: 'center',
+  overflow: 'hidden',
+});
+
+const SPIN_KEYFRAMES = Style.registerKeyframes({
```

Comment:
These widgets would be helpful elsewhere too; keeping them consistently used in the widgets dir or something